### PR TITLE
GitHub hooks setup

### PR DESCRIPTION
**Pull request by prakashchoudhary07**
Pre-commit, Pre-push, post-checkout hooks setup

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+echo "Linting your code, please wait⏳\n"
```
Comment (swarajpure) [inline .husky/pre-commit:4]: Shouldn't this be `Checking for linting problems, please wait`?
What do you think?

File: .husky/pre-push [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+. "$(dirname "$0")/_/husky.sh"
+
+echo "\nStarting test's, \nhold on tight!, let's  hope all goes smooth  ☠ ☠ \n"
```
Comment (swarajpure) [inline .husky/pre-push:4]: `tests` instead of `test's`?

File: package.json [hunk 3]
```diff
@@ -22,7 +22,8 @@
     "start": "ember serve",
     "test": "npm-run-all lint test:*",
     "test:ember": "ember test",
-    "test:ember:server": "ember test --server"
+    "test:ember:server": "ember test --server",
+    "prepare": "husky install"
```
Comment (swarajpure) [inline package.json:26]: smooth!

File: package.json [hunk 4]
```diff
@@ -57,6 +58,7 @@
     "eslint-plugin-ember": "^10.2.0",
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-prettier": "^3.3.1",
+    "husky": "^6.0.0",
```
Comment (swarajpure) [inline package.json:61]: We won't be using `pre-commit` anymore, right? Can we have it uninstalled? Also the script on line 76-77-78 can be removed 🙂

**Review by swarajpure — CHANGES_REQUESTED** [review]
Thanks for this PR!

Suggesting some changes, please check 🙂

File: package.json (hunk 4 above)
Comment (prakashchoudhary07) [reply package.json:61]: I thought of removing it after this PR getting accepted :sweat_smile:

Comment (prakashchoudhary07) [reply package.json:61]: Will remove it now

**Review by swarajpure — APPROVED** [review]
Looks good to me!

Thank you! 🎉

**Pull request merged**